vertx-auth-oauth2/src/main/java/io/vertx/ext/auth/oauth2/dcr/impl/KeycloakClientRegistrationImpl.java

```diff
@@ -0,0 +1,62 @@
+package io.vertx.ext.auth.oauth2.dcr.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.impl.http.SimpleHttpClient;
+import io.vertx.ext.auth.impl.http.SimpleHttpResponse;
+import io.vertx.ext.auth.oauth2.DCROptions;
+import io.vertx.ext.auth.oauth2.DCRRequest;
+import io.vertx.ext.auth.oauth2.DCRResponse;
+import io.vertx.ext.auth.oauth2.dcr.KeycloakClientRegistration;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public final class KeycloakClientRegistrationImpl implements KeycloakClientRegistration {
+
+  private final Vertx vertx;
+  private final SimpleHttpClient simpleHttpClient;
+
+  private final DCROptions dcrOptions;
+
+  public KeycloakClientRegistrationImpl(Vertx vertx, DCROptions dcrOptions) {
+    this.vertx = vertx;
+    this.dcrOptions = dcrOptions;
+    this.simpleHttpClient = new SimpleHttpClient(vertx, "dcr-client", dcrOptions.getHttpClientOptions());
+  }
+
+  @Override
+  public DCRResponse create(String clientId) {
```

Comment:
This is an async method (it performs some sort of IO) so we should return `Future<DCRResponse>`